GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/IPC/VFSForGitNotification.m

```diff
@@ -0,0 +1,142 @@
+    NSParameterAssert(error);
+    
+    id identifier = jsonMessage[IdentifierKey];
+    if (![identifier isKindOfClass:[NSNumber class]])
+    {
+        *error = @"Unexpected message id/format)";
+        return NO;
+    }
+    
+    Identifier idValue = [identifier integerValue];
+    switch (idValue)
+    {
+        case AutomountStart:
+        {
+            *notification = [[VFSForGitNotification alloc]
+                             initAsAutomountStartWithMessage:jsonMessage
+                             error:error];
+            break;
+        }
+        
+        case MountSuccess:
+        {
+            *notification = [[VFSForGitNotification alloc]
+                             initAsMountSuccessWithMessage:jsonMessage
+                             error:error];
+            break;
+        }
+            
+        case MountFailure:
+        {
+            *notification = [[VFSForGitNotification alloc]
+                             initAsMountFailureWithMessage:jsonMessage
+                             error:error];
+            break;
+        }
+            
+        default:
+        {
+            *notification = nil;
+            *error = @"Unrecognised message id";
+            break;
+        }
+    }
+    
+    return *notification != nil;
+}
+
+- (instancetype) initAsAutomountStartWithMessage:(NSDictionary *_Nonnull) messageDict
+                                           error:(NSString *_Nonnull *_Nonnull) error
+{
+    if (self = [super init])
+    {
+        id repoCount = messageDict[EnlistmentCountKey];
+        if (repoCount && [repoCount isKindOfClass:[NSNumber class]])
+        {
+            _title = [(NSString *) AutomountTitle copy];
+            _message = [[NSString stringWithFormat:
+                         (NSString *) AutomountStartMessageFormat,
+                         [repoCount unsignedIntegerValue]] copy];
+            return self;
+        }
+        
+        *error = @"Missing repos count in AutomountStart message";
+        self = nil;
+    }
+    
+    return self;
+}
+
+- (instancetype) initAsMountSuccessWithMessage:(NSDictionary *_Nonnull) messageDict
+                                         error:(NSString *_Nonnull *_Nonnull) error
+{
+    return self =
+    [self initAsMountWithMessage:messageDict
+                           title:(NSString *) AutomountTitle
+                   messageFormat:(NSString *) AutomountSuccessMessageFormat
+                           error:error];
+}
+
+- (instancetype) initAsMountFailureWithMessage:(NSDictionary *_Nonnull) messageDict
+                                         error:(NSString *_Nonnull *_Nonnull) error
+{
+    return self =
+    [self initAsMountWithMessage:messageDict
+                           title:(NSString *) AutomountTitle
+                   messageFormat:(NSString *) AutomountFailureMessageFormat
+                           error:error];
+}
+
+- (instancetype) initAsMountWithMessage:(NSDictionary *_Nonnull) messageDict
```

Comment:
Formatting: in ObjC usually we don't put a space between parameter type and parameter name:
initAsMountWithMessage:(NSDictionary *_Nonnull)messageDict